tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/Package/SdkCodeTool.cs

```diff
@@ -0,0 +1,473 @@
+            string? tspConfigPath,
+            [Description("Commit SHA of the 'azure-rest-api-specs' repository to use. Must be a valid 40-character Git SHA. Required when tspConfigPath is a local file path.")]
+            string? specCommitSha,
+            [Description("Full name of the repository in 'owner/repo' format. Example: 'Azure/azure-rest-api-specs'. Required when tspConfigPath is a local file path.")]
+            string? specRepoFullName,
+            [Description("Path to 'tsp-location.yaml'. Optional. May be left empty if running inside a local azure-rest-api-specs repository.")]
+            string? tspLocationPath,
+            [Description("Emitter options in key-value format. Optional. Example: 'package-version=1.0.0-beta.1'.")]
+            string? emitterOptions,
+            CancellationToken ct = default)
+        {
+            try
+            {
+                // Validate inputs
+                if (string.IsNullOrEmpty(tspConfigPath) && string.IsNullOrEmpty(tspLocationPath))
+                {
+                    return CreateFailureResponse("Both 'tspconfig.yaml' and 'tsp-location.yaml' paths aren't provided. At least one of them is required.");
+                }
+
+                // Handle tsp-location.yaml case
+                if (!string.IsNullOrEmpty(tspLocationPath))
+                {
+                    return await RunTspUpdate(tspLocationPath, ct);
+                }
+
+                // Handle tspconfig.yaml case
+                return await GenerateSdkFromTspConfigAsync(localSdkRepoPath, tspConfigPath, specCommitSha, specRepoFullName, emitterOptions, ct);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Error occurred while generating SDK");
+                return CreateFailureResponse($"An error occurred: {ex.Message}");
+            }
+        }
+
+        [McpServerTool(Name = "azsdk_package_build_code"), Description("Build SDK code for a specified project locally.")]
+        public async Task<DefaultCommandResponse> BuildSdkAsync(
+            [Description("Absolute path to the SDK project.")]
+            string packagePath,
+            CancellationToken ct = default)
+        {
+            try
+            {
+                logger.LogInformation($"Building SDK for project path: {packagePath}");
+
+                // Validate inputs
+                if (string.IsNullOrEmpty(packagePath) || !Directory.Exists(packagePath))
+                {
+                    return CreateFailureResponse("Invalid project path. Please provide a valid project path.");
+                }
+
+                // Return if the project is python project
+                if (packagePath.Contains("azure-sdk-for-python", StringComparison.OrdinalIgnoreCase))
```

Comment:
@LibbaLawrence with language services going away do we not have a place to put these as like static properties? I guess we could throw this stuff in `Configuration/Constants.cs` or just keep it hardcoded for now.